server/src/main/kotlin/org/web3j/openapi/server/config/ContractAddresses.kt

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2020 Web3 Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.web3j.openapi.server.config
+
+import org.web3j.abi.datatypes.Address
+
+data class ContractAddresses(
+    val addresses: Map<String, Address>?
```

Comment:
Why do you need this property if inherits from `HashMap` already?

Also shouldn't be a data class in that case, it will be mutable.

Reply:
`Or, if you need for the HK2 binder, I think should be better to inherit from HashMap<String, Address>.` This is what I understood from this.
So, I remove the inheritence ? Because we need the class for the binding. In the `binder`, if we specify the `Map::class.java`, we won't be able to specify the types of the key and value

Reply:
Yes of course inherit from Map, but the reason why is to remove the `val addresses` otherwise what's the point in inheriting if we are not going to use it as a map?

In the binder you cannot specify `Map::class.java` but `ContractAddresses::class.java` so then why would you need `val addresses`?

Reply:
Fixed